### PR TITLE
Fix some issues once again

### LINKED ISSUE #79: [bug]标签下的分页功能不可用
比如`研究生`这个标签，在点击第二页的时候产生的链接是
```
https://域名/tag%5c研究生%5cpage2.html
```

### LINKED ISSUE #81: 新的使用者
[link](http://qjkm.me/)

### LINKED ISSUE #82: 首页摘要 代码过长会溢出
[示意图](http://ww1.sinaimg.cn/large/c0264382gy1fjvq47jl1vj20wv0nb406.jpg)

### LINKED ISSUE #106: A problem
Now i find ，if i named same name article ，then，a problem knock on a door，issue：i create some directory,and different directory but same name files

### LINKED ISSUE #114: 添加使用者
[Knect的博客](https://iKnect.me/)

### LINKED ISSUE #117: 作者你好，自带的主题能否实现多种语言的代码高亮
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### LINKED ISSUE #121: Include other files
Currently it seems it is not possible to include a file, is that the case? Other
projects have this built in:

https://gohugo.io/functions/readfile

### PR DESCRIPTION
**Pull request by w568w**
This PR will:

1. fix #82 ;
2. fix #117 by upgrading `highlight.js` to the latest version;
    - *p.s.*: since the library `.js` file size is too large (~1 MB), it is splitted into an independent bundle.
3. close #121 by implementing template function support;
4. close #79 , a duplicate of #86 ; close #106, a dup of #107 .
5. close #81 and close #114 because both websites have been down for a while.

### CODE REVIEW

File: funcs.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+	themePath  string
+	publicPath string
+	currentCwd string
+	global     *GlobalConfig
+}
+
+func (ctx FuncContext) FuncMap() template.FuncMap {
+	return template.FuncMap{
+		"i18n":     ctx.I18n,
+		"readFile": ctx.ReadFile,
```
Comment (imeoer) [inline funcs.go:20]: It's better to use uppercase `ReadFile`, like other template variables?

Comment (w568w) [reply funcs.go:20]: As referred to in #121,  [Hugo](https://gohugo.io/functions/readfile/) uses lowerCamelCase for all functions in `FuncMap`, and UpperCamelCase for all functions in tplData (e.g. `os.ReadFile`, `.Get`). I was following this naming style.

**Comment by imeoer** [comment]
others LGTM!

**Comment by imeoer** [comment]
Hi @w568w, maybe we can publish the release binary using github action automatically, and update the [doc](https://github.com/InkProject/ink#quick-start).

**Comment by w568w** [comment]
> Hi @w568w, maybe we can publish the release binary using github action automatically, and update the [doc](https://github.com/InkProject/ink#quick-start).

Good idea. I will resolve that in #140 then.

**Pull request merged**